src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -287,48 +287,58 @@ public override Task<string> Script(DatabaseViewContext context, DatabaseInfo ob
         public string Detach(DetachDatabaseRequestParams detachParams)
         {
             var sqlScript = string.Empty;
-            ConnectionInfo connectionInfo = this.GetConnectionInfo(detachParams.ConnectionUri);
             using (var dataContainer = CreateDatabaseDataContainer(detachParams.ConnectionUri, detachParams.ObjectUrn, false, null))
```

Comment:
What about having the data container close the connection when it's disposed? That seems like a better pattern to avoid these kinds of mistakes from happening in the future.

Reply:
CDataContainer.Dispose already has a bunch of connection closing logic, but still seems to miss this one. I'll look into it.